ccmlib/node.py

```diff
@@ -1993,12 +1993,20 @@ def _get_row_cache_entries_from_info_output(info):
 
 
 def _grep_log_for_errors(log, distinct_errors=False, search_str=None, case_sensitive=True):
+    def make_pat_for_log_level(level):
+        flags = 0 if case_sensitive else re.IGNORECASE
```

Comment:
This will work but I don't like open coding the default value for flags.

It would be more future-compatible to write it like:
```python
    expr = rf'\b{level}\b'
    return re.compile(expr) if case_sensitive else re.compile(expr, flags=re.IGNORECASE)
```

Reply:
oops, I see it's already merged.
can be done as a follow up

Reply:
This was is the formal API default value, don't thing it's going to change anytime soon

https://docs.python.org/3/library/re.html#re.compile

Also this might be a bit clearer, if you think this safety is needed

```python
kwargs = {'flags': re.I} if ignore_case else {}
return re.compile(regex, **kwargs)
```

Reply:
> This was is the formal API default value, don't thing it's going to change anytime soon

Other than python changing its APIs quite radically between major versions, we should let it decide about the default value, not hard code it ourselves, even if it's documented (for this version).  E.g. it might change to None in a future version.

The kwargs approach is also viable...

Reply:
Sent https://github.com/scylladb/scylla-ccm/pull/392 as a followup